docs/source/onnx.rst

```diff
@@ -79,433 +84,154 @@ Then, you can run::
-      %3 : Tensor = onnx::Cast[to=9](%2)
-      %4 : Long(2, 3) = onnx::Loop(%loop_range, %3, %input_data), scope: LoopModel2/loop
-        block0(%i.1 : Long(), %cond : bool, %x.6 : Long(2, 3)):
-          %8 : Long(2, 3) = onnx::Add(%x.6, %i.1), scope: LoopModel2/loop
-          %9 : Tensor = onnx::Cast[to=9](%2)
-          -> (%9, %8)
-      return (%4)
-
-The dynamic control flow is captured correctly. We can verify in backends with different loop range. ::
-
-    import caffe2.python.onnx.backend as backend
-    import numpy as np
-    import onnx
-    model = onnx.load('loop.onnx')
-
-    rep = backend.prepare(model)
-    outputs = rep.run((dummy_input.numpy(), np.array(9).astype(np.int64)))
-    print(outputs[0])
-    #[[37 37 37]
-    # [37 37 37]]
-
-
-    import onnxruntime as ort
-    ort_sess = ort.InferenceSession('loop.onnx')
-    outputs = ort_sess.run(None, {'input_data': dummy_input.numpy(),
-                                  'loop_range': np.array(9).astype(np.int64)})
-    print(outputs)
-    #[array([[37, 37, 37],
-    #       [37, 37, 37]], dtype=int64)]
-
-
-To avoid exporting a variable scalar tensor as a fixed value constant as part of the ONNX model, please
-avoid use of ``torch.Tensor.item()``. Torch supports implicit cast of single-element tensors to numbers.
-E.g.: ::
-
-    class LoopModel(torch.nn.Module):
-        def forward(self, x, y):
-            res = []
-            arr = x.split(2, 0)
-            for i in range(int(y)):
-                res += [arr[i].sum(0, False)]
-            return torch.stack(res)
-
-    model = torch.jit.script(LoopModel())
-    inputs = (torch.randn(16), torch.tensor(8))
-
-    out = model(*inputs)
-    torch.onnx.export(model, inputs, 'loop_and_list.onnx', opset_version=11, example_outputs=out)
-
-
-Type Annotations
---------------------------------
-TorchScript only supports a subset of Python types. You can find more details about type annotation
-`here <https://pytorch.org/docs/stable/jit_language_reference.html#id8>`_.
-
-Due to optimization purposes, TorchScript only supports variables with single static types for script functions.
-By default, each variable is assumed to be Tensor. If an argument to a ScriptModule function is not Tensor,
-its type should be specified using MyPy-style annotations.
-
-::
-
-    import torch
-
-    class Module(torch.nn.Module):
-        def forward(self, x, tup):
-            # type: (int, Tuple[Tensor, Tensor]) -> Tensor
-            t0, t1 = tup
-            return t0 + t1 + x
-
-If the type annotation is not specified, TorchScript compiler fails with the runtime error below.
-
-::
-
-  RuntimeError:
-  Tensor (inferred) cannot be used as a tuple:
-    File <filename>
-          def forward(self, x, tup):
-              t0, t1 = tup
-                       ~~~ <--- HERE
-              return t0 + t1 + x
-
+Before a model can be exported, it must be compiled into a :class:`torch.jit.ScriptModule`.
```

Comment:
Rephrase? The user does not need to provide a `torch.jit.ScriptModule` to export, although he can.